Client/Frontend/Fakespot/FakespotViewController.swift

```diff
@@ -96,6 +114,22 @@ class FakespotViewController: UIViewController, Themeable, Notifiable, UIAdaptiv
                 self?.updateContent()
             }
         }
+
+        viewModel.onAnalysisStatusChange = { [weak self] in
```

Comment:
Updating the content using this block goes against how we build up the content for the bottom sheet otherwise. I wonder if we can do this differently by maybe using our `updateContent` or creating the `needsAnalysisView` similar to the `FakespotNoAnalysisCardView` which shows either the button or the spinner depending on its state.

Reply:
I'm looking into something that will not involve removing and adding progress. That might help.

Reply:
The problem is at some point we have to remove the view, so we still need a reference to it. And doing that declarative with a viewElement array will involve removing all the elements in the stackview and readding them again, because we don't have a diffing mechanism.

Reply:
Yes, we should look into optimizing the code after MVP. Do you mind creating a task for this capturing your ideas on how to change it?